library/oidc.c

```diff
@@ -16,16 +16,23 @@
 #include <assert.h>
 #include <json.h>
 #include <sodium.h>
+#include <ctype.h>
 #include "ziti/ziti_log.h"
 #include "utils.h"
 #include "ziti/errors.h"
+#include "ziti/ziti_buffer.h"
+#include "ziti/ziti_model.h"
 
-#define code_len 8
+#define code_len 40
 #define code_verifier_len sodium_base64_ENCODED_LEN(code_len, sodium_base64_VARIANT_URLSAFE_NO_PADDING)
 #define code_challenge_len sodium_base64_ENCODED_LEN(crypto_hash_sha256_BYTES, sodium_base64_VARIANT_URLSAFE_NO_PADDING)
 
-#define default_cb_url "http://localhost:18889/auth/callback"
-#define default_client_id "native"
+#define _str(x) #x
+#define auth_cb_port 20314 /* 'OZ' */
+#define auth_url_path "/auth/callback"
+#define cb_url(host,port,path) "http://" host ":" _str(port) path
+#define default_cb_url cb_url("localhost",auth_cb_port,auth_url_path)
```

Comment:
how do you feel about using 127.0.0.1 not localhost? we've seen enough ipv4/v6 issues to make me wonder

Reply:
auth callback stuff is subject to change depending on if/how it works in mobile apps